File: Tema 10/ListadoPersonasClean/Domain/Entities/Persona.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

namespace Domain.Entities
{
    public class Persona
    {
        #region Atributos privados
        private int _id;
        private string _nombre;
        private string _apellidos;
        private DateTime _fechaNacimiento;
        private string _direccion;
        private string _telefono;
        private string _foto;
        private int _idDepartamento;
        #endregion

        #region Getters y Setters
        [Required(ErrorMessage = "Campo obligatorio")]
        public int Id
        {
            get { return _id; }
            set { _id = value; }
        }

        [Required(ErrorMessage = "Campo obligatorio")]
        public string Nombre
        {
            get { return _nombre; }
            set { _nombre = value; }
        }

        [Required(ErrorMessage = "Campo obligatorio")]
        public string Apellidos
        {
            get { return _apellidos; }
            set { _apellidos = value; }
        }

        [StringLength(10)]
        public DateTime FechaNacimiento
        {
            get { return _fechaNacimiento; }
            set { _fechaNacimiento = value; }
        }

        [Display(Name = "Direccion")]
        public string Direccion
        {
            get { return _direccion; }
            set { _direccion = value; }
        }

        [StringLength(12)]
        public string Telefono
        {
            get { return _telefono; }
            set { _telefono = value; }
        }

        [Display(Name = "Foto")]
        public string? Foto
        {
            get { return _foto; }
            set { _foto = value; }
        }

        [Display(Name = "Departamento")]
        public int? IdDepartamento
        {
            get { return _idDepartamento; }
            set { _idDepartamento = value ?? 0; }
        }
        #endregion

        #region Constructores
        public Persona() { }

        public Persona(int id, string nombre, string apellidos, DateTime fechaNacimiento, string direccion, string telefono, string foto, int idDepartamento)
        {
            _id = id;
            _nombre = nombre;
            _apellidos = apellidos;
            _fechaNacimiento = fechaNacimiento;
            _direccion = direccion;
            _telefono = telefono;
            _foto = foto;
            _idDepartamento = idDepartamento;
        }
        #endregion
    }
}

File: Tema15/ListadoPersonasAngular/src/Domain/Entities/Persona.ts
export class Persona {
  private _id: number;
  private _nombre: string;
  private _apellidos: string;
  private _fechaNacimiento: Date;
  private _direccion: string;
  private _telefono: string;
  private _foto: string;
  private _idDepartamento: number;

  public constructor(id: number = 0, nombre: string = '', apellidos: string = '', fechaNacimiento: Date = new Date(), direccion: string = '', telefono: string = '', foto: string = '', idDepartamento: number = 0) {
    this._id = id;
    this._nombre = nombre;
    this._apellidos = apellidos;
    this._fechaNacimiento = fechaNacimiento;
    this._direccion = direccion;
    this._telefono = telefono;
    this._foto = foto;
    this._idDepartamento = idDepartamento;
  }

  public get id(): number { 
    return this._id; 
  }

  public set id(value: number) { 
    this._id = value; 
  }

  public get nombre(): string { 
    return this._nombre; 
  }

  public set nombre(value: string) { 
    this._nombre = value; 
  }

  public get apellidos(): string { 
    return this._apellidos; 
  }

  public set apellidos(value: string) { 
    this._apellidos = value; 
  }

  public get fechaNacimiento(): Date { 
    return this._fechaNacimiento; 
  }

  public set fechaNacimiento(value: Date) { 
    this._fechaNacimiento = value; 
  }

  public get direccion(): string { 
    return this._direccion; 
  }
  
  public set direccion(value: string) { 
    this._direccion = value; 
  }

  public get telefono(): string { 
    return this._telefono; 
  }

  public set telefono(value: string) { 
    this._telefono = value; 
  }

  public get foto(): string { 
    return this._foto; 
  }

  public set foto(value: string) { 
    this._foto = value; 
  }

  public get idDepartamento(): number { 
    return this._idDepartamento; 
  }

  public set idDepartamento(value: number) { 
    this._idDepartamento = value; 
  }
}